src/aselo-webchat.ts

```diff
@@ -0,0 +1,207 @@
+  startEngagementOnInit: false,
+  preEngagementConfig: {
+    description: "Let's get started",
+    fields:
+      [{
+        label: "What is your helpline?",
+        type: "SelectItem",
+        attributes:
+        {
+          name: "helpline",
+          required: true,
+          readOnly: false
+        },
+        options: [
+          {
+            value: "Select helpline",
+            label: "Select helpline",
+            selected: true
+          },
+          {
+            value: "Fake Helpline",
+            label: "Fake Helpline",
+            selected: false
+          },
+        ]
+      }],
+    submitLabel: "Let's chat!"
+  }
+};
+
+const mapHelplineLanguage = (helpline: string) => {
+  switch (helpline) {
+    case 'Fake Helpline':
+      return 'dk';
+    default:
+      return defaultLanguage;
+  }
+}
+
+const getChangeLanguageWebChat = (manager: FlexWebChat.Manager) => (language: string) => {
+  const twilioStrings = { ...manager.strings }; // save the originals
+  const setNewStrings = (newStrings: FlexWebChat.Strings) => (manager.strings = { ...manager.strings, ...newStrings });
+  const translationErrorMsg = 'Could not translate, using default';
+  
+  try {
+    if (language !== defaultLanguage && translations[language]) {
+      setNewStrings({ ...twilioStrings, ...translations[defaultLanguage], ...translations[language] });
+    } else {
+      setNewStrings({ ...twilioStrings, ...translations[defaultLanguage] });
+    }
+
+    console.log('Translation OK');
+  } catch (err) {
+    window.alert(translationErrorMsg);
+    console.error(translationErrorMsg, err);
+    getChangeLanguageWebChat(manager)(defaultLanguage);
+  }
+}
+
+type ChannelAndManagerFn = (channel: Channel, manager: FlexWebChat.Manager) => void;
+
+const doWithChannel = (callback: ChannelAndManagerFn) => (manager: FlexWebChat.Manager) => {
+  const { channelSid } = manager.store.getState().flex.session;
+    manager
+      .chatClient.getChannelBySid(channelSid)
+      .then(channel => {
+        callback(channel, manager);
+      });
+}
+
+const unlockInput = (manager: FlexWebChat.Manager) => {
+  const { user } = manager.chatClient;
+  const { lockInput, ...attributes } = user.attributes as any;
+  user.updateAttributes(attributes);
+}
+
+// TODO: we shouldn't need to cast a Channel to EventEmitter
+type EventEmitter = {
+  once: (eventName: string, listener: Function) => void;
+};
+
+const setListenerToUnlockInput = async (channel: Channel, manager: FlexWebChat.Manager) => {
+  if (!channel) return;
+
+  const cb = () => {
+    // Re-enable input
+    unlockInput(manager)
+  }
+
+  // User is not alone in the channel (possible cause to enter this case is page reload)
+  const membersCount = await channel.getMembersCount();
+  if (membersCount > 1) {
+    cb();
+    return;
+  }
+  
+  // Adds an event listener that will run only once
+  (channel as Channel & EventEmitter).once("memberJoined", () => {
```

Comment:
This compiles good for me without the casting. Are you still receiving the error?

![Screenshot from 2021-02-12 11-14-18](https://user-images.githubusercontent.com/15805319/107779050-9cfc5e80-6d23-11eb-9620-ceec3bddebed.png)

Reply:
great, it's working without the cast!